uaclient/tests/test_cli_status.py

```diff
@@ -975,6 +989,45 @@ def contract_info_side_effect(cfg, token):
 
         assert output["errors"][0]["message"] == warning_message
 
+    @pytest.mark.parametrize(
+        "contract_changed,is_attached",
+        (
+            (False, True),
+            (True, False),
+            (True, True),
+            (False, False),
+        ),
+    )
+    @mock.patch(M_PATH_UACONFIG + "add_notice")
+    def test_is_contract_changed(
+        self,
+        m_add_notice,
+        _m_getuid,
+        _m_get_contract_information,
+        _m_get_available_resources,
+        _m_should_reboot,
+        _m_remove_notice,
+        _m_contract_changed,
+        contract_changed,
+        is_attached,
+        capsys,
+        FakeConfig,
+    ):
+        _m_contract_changed.return_value = contract_changed
+        cfg = FakeConfig().for_attached_machine()
+        action_status(
+            mock.MagicMock(all=False, simulate_with_token=None), cfg=cfg
+        )
+        if contract_changed:
+            if is_attached:
```

Comment:
That variable here is not being used to set if the `cfg` module should behave as attached or not. It seems that test is always using a cfg that is attached. Is that right ?

Reply:
Yeah, let me check for both conditions.